src/services/servicesRequests.js

```diff
@@ -0,0 +1,333 @@
+import { courtServerEndpoint } from '../endpoints'
+import {
+  FORBIDDEN_RESPONSE,
+  UNAUTHORIZED_RESPONSE,
+  USER_NOT_FOUND,
+} from './responseCodes'
+
+const COURT_SERVER_ENDPOINT = courtServerEndpoint()
+
+// https://court-backend-usability.eth.aragon.network
+
+export async function createSession(address, signature, timestamp) {
+  const endpoint = `${COURT_SERVER_ENDPOINT}/users/${address}/sessions`
+
+  try {
+    const rawResponse = await fetch(endpoint, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ signature, timestamp }),
+    })
+
+    const jsonResponse = await rawResponse.json()
+
+    if (rawResponse.ok) {
+      return {
+        authenticated: jsonResponse?.authenticated,
+        error: null,
+      }
+    }
+
+    const errors = jsonResponse.errors
+      .map(err => Object.values(err).join(', '))
+      .join(', ')
+
+    return {
+      error: errors,
+    }
+  } catch (err) {
+    console.error(err)
+    return { error: err }
+  }
+}
+
+export async function subscribeToNotifications(address, email) {
+  const endpoint = `${COURT_SERVER_ENDPOINT}/users/${address}/email`
+
+  try {
+    const rawResponse = await fetch(endpoint, {
+      method: 'PUT',
+      credentials: 'include',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ email }),
+    })
+
+    const jsonResponse = await rawResponse.json()
+
+    if (rawResponse.ok) {
+      return {
+        subscribedEmail: jsonResponse?.email,
+        sent: jsonResponse?.sent,
+        error: null,
+      }
+    }
+
+    const errors = jsonResponse.errors
+      .map(err => Object.values(err).join(', '))
+      .join(', ')
+
+    return {
+      needsSignature:
```

Comment:
It looks like these error and authentication validation bits could be taken out and abstracted into a function, e.g.

```js
function notificationsApi(req) {
  const rawResponse = await req()

  if (!rawResponse.ok) {
    const errors = jsonResponse.errors
      .map(err => Object.values(err).join(', '))
      .join(', ')

    return {
      needsSignature:
        rawResponse.status === UNAUTHORIZED_RESPONSE ||
        rawResponse.status === USER_NOT_FOUND ||
        rawResponse.status === FORBIDDEN_RESPONSE,
      error: errors,
    }
  }

  return rawResponse
}